src/shogun/multiclass/KNN.cpp

```diff
@@ -123,33 +124,53 @@ SGMatrix<index_t> KNN::nearest_neighbors()
 	distance->precompute_lhs();
 	distance->precompute_rhs();
 
-	//for each test example
-	for (auto i : SG_PROGRESS(range(n)))
+	switch (knn_solver)
 	{
-		COMPUTATION_CONTROLLERS
-		//lhs idx 0..num train examples-1 (i.e., all train examples) and rhs idx i
-		distances_lhs(dists,0,m_train_labels.vlen-1,i);
+	case KNN_BRUTE:
+	{
+		//for each test example
+		for (auto i : SG_PROGRESS(range(n)))
+		{
+			COMPUTATION_CONTROLLERS
+			//lhs idx 0..num train examples-1 (i.e., all train examples) and rhs idx i
+			distances_lhs(dists,0,m_train_labels.vlen-1,i);
 
-		//fill in an array with 0..num train examples-1
-		for (int32_t j=0; j<m_train_labels.vlen; j++)
-			train_idxs[j]=j;
+			//fill in an array with 0..num train examples-1
+			for (int32_t j=0; j<m_train_labels.vlen; j++)
+				train_idxs[j]=j;
 
-		//sort the distance vector between test example i and all train examples
-		Math::qsort_index(dists.vector, train_idxs.vector, m_train_labels.vlen);
+			//sort the distance vector between test example i and all train examples
+			Math::qsort_index(dists.vector, train_idxs.vector, m_train_labels.vlen);
 
 #ifdef DEBUG_KNN
-		io::print("\nQuick sort query {}\n", i);
-		for (int32_t j=0; j<m_k; j++)
-			io::print("{} ", train_idxs[j]);
-		io::print("\n");
+			io::print("\nQuick sort query {}\n", i);
+			for (int32_t j=0; j<m_k; j++)
+				io::print("{} ", train_idxs[j]);
```

Comment:
use SG_DEBUG instead of io::print

Reply:
Loop could be replaced with SGVector to_string no?

Reply:
Or should I use the `display_vector() `function?

Reply:
no `display_vector`, but do what was suggested above